core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -572,11 +572,14 @@ object DocumentRenderer extends Renderer[Document] {
   }
 
   private lazy val directiveRenderer: Renderer[Directive] = new Renderer[Directive] {
-    override def unsafeRender(d: Directive, indent: Option[Int], writer: StringBuilder): Unit = {
-      writer append '@'
-      writer append d.name
-      inputArgumentsRenderer.unsafeRender(d.arguments, indent, writer)
-    }
+    override def unsafeRender(d: Directive, indent: Option[Int], writer: StringBuilder): Unit =
+      if (d.isIntrospectable) {
+        writer append '@'
+        writer append d.name
+        inputArgumentsRenderer.unsafeRender(d.arguments, indent, writer)
+      } else {
+        writer.deleteCharAt(writer.size - 1)
```

Comment:
Is this to delete the potential space before it the directive?

Reply:
Yeap

Reply:
I think I might have been a bit lazy here, there might be a better way to do it perhaps (by not adding the space to begin with?). I couldn't find where the space is added though :/

Reply:
I would use the `when` combinator instead since this isn't handling the "compact" mode case

Reply:
Hmm seems that `when` doesn't work because we still get the empty strings. I used `filter` within the `contramap` method to filter out the directives from the List instead 👍